src/mixins/labels/coordinateMeasurement.js

```diff
@@ -0,0 +1,145 @@
+import * as THREE from "three";
+
+import { COORDINATE_LABELS_GROUP_NAME } from "../../enums";
+import { CoordinateLabelsMixin } from "./coordinateLabels";
+
+/*
+ * Mixin containing the logic for coordinate measurements.
+ * Handles selection, label creation, and clipboard operations for coordinate measurements.
+ */
+export const CoordinateMeasurementMixin = (superclass) =>
+    class extends CoordinateLabelsMixin(superclass) {
+        constructor(config) {
+            super(config);
+            this.coordinateMeasurementGroup = new THREE.Group();
+            this.coordinateMeasurementGroup.name = COORDINATE_LABELS_GROUP_NAME;
+            this.selectedAtomsForCoordinates = new Set();
+            this.coordinatesArray = [];
+            this.isCoordinateMeasurementActive = false;
+            this.structureGroup.add(this.coordinateMeasurementGroup);
+        }
+
+        /**
+         * Toggles coordinate measurement mode on/off
+         */
+        toggleCoordinateMeasurement() {
+            this.isCoordinateMeasurementActive = !this.isCoordinateMeasurementActive;
+            if (!this.isCoordinateMeasurementActive) {
+                this.clearCoordinateMeasurements();
+            }
+            this.coordinateMeasurementGroup.visible = this.isCoordinateMeasurementActive;
+            this.render();
+        }
+
+        /**
+         * Copies the current array of coordinates to clipboard
+         */
+        copyCoordinatesToClipboard() {
+            if (this.coordinatesArray.length === 0) return;
+            const coordsText =
+                this.coordinatesArray.length === 1
+                    ? JSON.stringify(this.coordinatesArray[0])
+                    : JSON.stringify(this.coordinatesArray);
+
+            navigator.clipboard.writeText(coordsText).catch(console.error);
+        }
+
+        /**
+         * Selects or deselects an atom for coordinate measurement
+         * @param {THREE.Mesh} atom - The atom to toggle selection for
+         */
+        toggleAtomCoordinateSelection(atom) {
+            if (!this.isCoordinateMeasurementActive) return;
+
+            if (this.selectedAtomsForCoordinates.has(atom.uuid)) {
+                this.deselectAtomCoordinate(atom);
+            } else {
+                this.selectAtomCoordinate(atom);
+            }
+            this.copyCoordinatesToClipboard();
+            this.render();
+        }
+
+        /**
+         * Selects an atom and creates its coordinate label
+         * @param {THREE.Mesh} atom - The atom to select
+         */
+        selectAtomCoordinate(atom) {
+            this.selectedAtomsForCoordinates.add(atom.uuid);
```

Comment:
Is probably also duplicate with the other measurements - should have the measuments base mixin, measurementsHolder and reuse